git_wrapper/tag.py

```diff
@@ -0,0 +1,51 @@
+#! /usr/bin/env python
+"""This module acts as an interface for acting on git tags"""
+
+from git_wrapper import exceptions
+from git_wrapper.utils.decorators import reference_exists
+
+
+class GitTag(object):
+
+    def __init__(self, git_repo, logger):
+        """Constructor for GitTag object
+
+            :param repo.GitRepo git_repo: An already constructed GitRepo object
+            :param logging.Logger logger: A pre-configured Python Logger object
+        """
+        self.git_repo = git_repo
+        self.logger = logger
+
+    @reference_exists('reference')
+    def create(self, name, reference):
+        """Create a new tag to the target reference.
+
+           :param str name: New tag's name
+           :param str reference: What the tag should point to
+        """
+        self.git_repo.repo.create_tag(name, reference)
```

Comment:
I don't see anything obvious beyond references not existing, which the decorator covers but it would be safer to add some error handling just in case anyway. I will do that.